src/handlers/next/create-app-register-handler.ts

```diff
@@ -13,6 +14,39 @@ import { validateAllowSaleorUrls } from "./validate-allow-saleor-urls";
 
 const debug = createDebug("createAppRegisterHandler");
 
+type HookCallbackErrorParams = {
+  status?: number;
+  body?: object;
+  message?: string;
+};
+
+class HookCallbackError extends Error {
```

Comment:
```suggestion
class RegisterCallbackError extends Error {
```
Dont have a strong preference, but probably this name would be more clear for me if seen the first time in the logs.

Reply:
Especially if fn to create one is called createCallbackError